src/cryptonote_core/tx_pool.cpp

```diff
@@ -917,13 +939,20 @@ namespace cryptonote
 
     for (const auto& image : key_images)
     {
-      spent.push_back(m_spent_key_images.find(image) == m_spent_key_images.end() ? false : true);
+      bool is_spent = false;
+      const auto found = m_spent_key_images.find(image);
+      if (found != m_spent_key_images.end())
+      {
+        for (const crypto::hash& tx_hash : found->second)
+          is_spent |= m_blockchain.txpool_tx_matches_category(tx_hash, relay_category::broadcasted);
+      }
```

Comment:
This behavior change was intentional, although I'm not sure its correct. This function is only used by the ZMQ rpc code to return whether a list of key-images are spent. This is now hiding `do_not_relay` txes from the ZMQ rpc.

Is the ZMQ rpc considered `!restricted`? I think it is actually ... ?

Reply:
I'm wondering "out loud" here because it does seem funky to "hide" a key-image from a ZMQ rpc call that could've just provided it. OTOH, it doesn't seem natural to respond to `do_not_relay` txes but not stem, etc.

Reply:
I think it's unrestricted in practice since only the first RPC code uses that variable. You may choose to add a restricted mode later or not if it make sense.

Reply:
So this currently uses `relay_category::broadcasted` still, which will now filter out `do_not_relay` transactions. This is for the ZMQ interface only so impact should be minimal.